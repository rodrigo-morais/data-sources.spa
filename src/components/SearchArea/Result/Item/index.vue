<template src='./Item.html'></template>

<script src='./Item.js'></script>

<style scoped src='./Item.css'></style>
