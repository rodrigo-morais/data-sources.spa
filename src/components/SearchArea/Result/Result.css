.result {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: 1fr 1fr;

  grid-column: span 2;
  grid-row: 2;
}
