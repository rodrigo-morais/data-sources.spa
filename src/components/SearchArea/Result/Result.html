<div class="result" v-if="hasData">
  <Item
    v-for="(item, index) in dataSources"
    :name="item.name"
    :archived="item.archived"
   />
</div>
