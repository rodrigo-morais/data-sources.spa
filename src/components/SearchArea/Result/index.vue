<template src='./Result.html'></template>

<script src='./Result.js'></script>

<style scoped src='./Result.css'></style>
