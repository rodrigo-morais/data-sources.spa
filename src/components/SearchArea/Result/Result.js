import Vue from 'vue'
import { mapState } from 'vuex'
import Component from 'vue-class-component'
import Item from './Item/index.vue'

export default Component({
  name: 'SearchResult',
  components: { Item },
  computed: {
    ...mapState({
      dataSources: ({ search: { data } }) => (data ? data.dataSources : []),
      hasData: ({ search: { data, loading, error } }) =>
        data && !loading && !error && data.dataSources,
    }),
  },
})(class SearchResult extends Vue {})
