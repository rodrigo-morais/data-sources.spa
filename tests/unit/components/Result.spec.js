import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import Result from '@/components/SearchArea/Result/index.vue'


const localVue = createLocalVue()
localVue.use(Vuex)

const actions = { fetchDataSources: jest.fn() }
const state = {
  search: {
    loading: false,
    data: { dataSources: [{ name: 'DataSource 1', archived: true }, { name: 'DataSource 2', archived: false }] },
    error: false,
  },
}
const store = new Vuex.Store({
  state,
  actions,
})

describe('Result.vue', () => {
  it('renders according to design', () => {
    const renderer = createRenderer()
    const wrapper = shallowMount(Result, { store, localVue })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
