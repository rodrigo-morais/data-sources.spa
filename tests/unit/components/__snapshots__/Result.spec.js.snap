// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result.vue renders according to design 1`] = `
<div class="result">
  <Item-stub name="DataSource 1" archived="true"></Item-stub>
  <Item-stub name="DataSource 2"></Item-stub>
</div>
`;
